column_diagnostics/column_diagnostics.F90

```diff
@@ -64,17 +64,28 @@ module column_diagnostics_mod
           column_diagnostics_header,   &
           close_column_diagnostics_units
 
+
+interface initialize_diagnostic_columns
+  module procedure initialize_diagnostic_columns_r4
+  module procedure initialize_diagnostic_columns_r8
+end interface initialize_diagnostic_columns
+
+interface column_diagnostics_header
+  module procedure column_diagnostics_header_r4
+  module procedure column_diagnostics_header_r8
+end interface column_diagnostics_header
+
 !private
 
 !--------------------------------------------------------------------
 !----    namelist -----
 
-real          :: crit_xdistance = 4.0 !< model grid points must be within crit_xdistance in
+real(kind=r8_kind) :: crit_xdistance = 4.0 !< model grid points must be within crit_xdistance in
```

Comment:
I think we should avoid using the d for exponents since its not as portable. Some compilers count the doubles as double whatever the default is so this would end up being 16 bytes with gcc for a normal compilation.